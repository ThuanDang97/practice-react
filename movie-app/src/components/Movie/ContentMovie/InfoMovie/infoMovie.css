.info_movie {
  display: flex;
}

.info_movie .info_wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 40px;
}

.tittle {
  display: flex;
  flex-direction: column;
}

.tittle h2 {
  width: 100%;
  margin: 0;
  padding: 0;
  font-size: 2.2rem;
}

.tittle h2 a {
  font-weight: 700;
  color: white;
}

.tittle h2 .release_date {
  opacity: 0.8;
  font-weight: 400;
  margin-left: 10px;
  color: white;
}

.facts {
  display: flex;
}

.facts .genres a {
  color: white;
  font-size: 18px;
}

.facts .genres a:hover {
  opacity: 0.8;
}

.genres a + a {
  padding-left: 10px;
}

.runtime {
  color: white;
  font-size: 18px;
  padding-left: 20px;
  position: relative;
  top: 0;
  left: 0;
}

.runtime::before {
  font-size: 1.1em;
  line-height: 1;
  content: '\2022';
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 7px;
  display: inline-flex;
  align-content: center;
  align-items: center;
  z-index: -1;
}

.overview {
  width: 100%;
  display: flex;
  flex-direction: column;
}

.overview h3 {
  width: 100%;
  margin: 0 0 8px 0;
  font-weight: 600;
  font-size: 28px;
  color: white;
}

.overview .desc {
  font-size: 18px;
  color: white;
  line-height: 27px;
}

.options {
  margin-bottom: 20px;
  width: 100%;
  height: 68px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  list-style: none;
  padding: 0;
}

.details .outer_ring {
  width: 65px;
  height: 65px;
}
.chart {
  display: flex;
  align-items: center;
  margin-right: 20px;
}

.chart .text {
  font-weight: 800;
  margin-left: 10px;
  white-space: pre-line;
  color: white;
}

.option-single {
  margin-right: 20px;
  background: rgba(3, 37, 65, 1);
  border-radius: 50%;
  width: 40px;
  height: 40px;
  box-sizing: border-box;
  cursor: pointer;
}

.option-single i {
  display: flex;
  justify-content: center;
}

.option-single img {
  width: 20px;
  height: 40px;
}
