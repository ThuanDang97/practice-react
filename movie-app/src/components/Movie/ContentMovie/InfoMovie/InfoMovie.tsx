import './infoMovie.css';
import Tooltip from '../../../Tooltip/Tooltip';
import { Genre } from '../../../../interfaces/Movie/IMovieInfo';
import { options } from '../../../../constants/MockData';
import { convertMinsToHrsMins } from '../../../../helper';
import { convertRating } from '../../../../utils';

export type TProps = {
  title: string;
  releaseDate: string;
  genres: Genre[];
  runtime: number;
  overview: string;
  scoreVoteAverage: number;
};

const InfoMovie = (props: TProps): JSX.Element => {
  const { title, releaseDate, genres, runtime, overview, scoreVoteAverage } = props;
  const time = convertMinsToHrsMins(runtime!);
  const score = convertRating(scoreVoteAverage);

  return (
    <section className="info_movie">
      <div className="info_wrapper">
        <div className="tittle">
          <h2>
            <a href="#" data-testid="title">
              {title}
            </a>
            <span className="tag release_date" data-testid="release">
              ({releaseDate?.split('', 4)})
            </span>
          </h2>
          <div className="facts">
            <span className="genres" data-testid="genres">
              {genres?.map((genre) => (
                <a href="#" key={genre.id}>
                  {genre.name}
                </a>
              ))}
            </span>
            <span className="runtime" data-testid="runtime">
              {time}
            </span>
          </div>
        </div>
        <ul className="options">
          <li className="chart">
            <div className="details">
              <div className="outer_ring">
                <div className="user_score_chart">
                  <div className="percent">
                    <span className="icon icon-r73" data-testid="score">
                      {score}
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div className="text">
              User
              <br />
              Score
            </div>
          </li>
          {options.map((option) => (
            <Tooltip text={option.text} key={option.id}>
              <li className="option-single">
                <i>
                  <img src={option.src} alt={option.alt} />
                </i>
              </li>
            </Tooltip>
          ))}
        </ul>
        <div className="overview">
          <h3>Overview</h3>
          <p className="desc" data-testid="desc">
            {overview}
          </p>
        </div>
      </div>
    </section>
  );
};
export { InfoMovie };
