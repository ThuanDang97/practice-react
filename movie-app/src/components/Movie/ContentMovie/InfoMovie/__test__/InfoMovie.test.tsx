import { screen, render } from '@testing-library/react';
import { InfoMovie, TProps } from '../InfoMovie';

describe('Test Button component', () => {
  const mockData: TProps = {
    title: 'End Game',
    releaseDate: '2020-10-10',
    genres: [
      {
        id: 1,
        name: 'Action',
      },
    ],
    runtime: 120,
    overview: 'Movie over view',
    scoreVoteAverage: 7.5,
  };

  test('Component Info Movie matches DOM Snapshot', () => {
    const component = <InfoMovie {...mockData} />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Info Movie should render correctly', () => {
    render(<InfoMovie {...mockData} />);

    expect(screen.getByTestId('title')).toHaveTextContent('End Game');
    expect(screen.getByTestId('release')).toHaveTextContent('2020');
    expect(screen.getByTestId('genres')).toHaveTextContent('Action');
    expect(screen.getByTestId('runtime')).toHaveTextContent('2h 0m');
    expect(screen.getByTestId('score')).toHaveTextContent('75');
    expect(screen.getByTestId('desc')).toHaveTextContent('Movie over view');
  });
});
