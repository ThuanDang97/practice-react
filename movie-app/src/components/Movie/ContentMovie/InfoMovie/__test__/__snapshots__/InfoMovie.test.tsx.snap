// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Component Info Movie matches DOM Snapshot 1`] = `
<div>
  <section
    class="info_movie"
  >
    <div
      class="info_wrapper"
    >
      <div
        class="tittle"
      >
        <h2>
          <a
            data-testid="title"
            href="#"
          >
            End Game
          </a>
          <span
            class="tag release_date"
            data-testid="release"
          >
            (
            2
            0
            2
            0
            )
          </span>
        </h2>
        <div
          class="facts"
        >
          <span
            class="genres"
            data-testid="genres"
          >
            <a
              href="#"
            >
              Action
            </a>
          </span>
          <span
            class="runtime"
            data-testid="runtime"
          >
            2h 0m
          </span>
        </div>
      </div>
      <ul
        class="options"
      >
        <li
          class="chart"
        >
          <div
            class="details"
          >
            <div
              class="outer_ring"
            >
              <div
                class="user_score_chart"
              >
                <div
                  class="percent"
                >
                  <span
                    class="icon icon-r73"
                    data-testid="score"
                  >
                    75
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="text"
          >
            User
            <br />
            Score
          </div>
        </li>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div>
            <li
              class="option-single"
            >
              <i>
                <img
                  alt="Add to list"
                  src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-159-thumbnails-list-white-c260ea972eebf812289fd3c41d0d2c1dff33ecbcd62be13fba8eeaf9de173789.svg"
                />
              </i>
            </li>
          </div>
          <span
            class="tooltip-text"
          >
            Add to list
          </span>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div>
            <li
              class="option-single"
            >
              <i>
                <img
                  alt="Favorite"
                  src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-13-heart-white-28d2cc2d6418c5047efcfd2438bfc5d109192671263c270993c05f130cc4584e.svg"
                />
              </i>
            </li>
          </div>
          <span
            class="tooltip-text"
          >
            Remove from your favorite list
          </span>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div>
            <li
              class="option-single"
            >
              <i>
                <img
                  alt="Watchlist"
                  src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-73-bookmark-white-432e98d550b7e4c80b06272c49475b0db85a60f6fae450420713004b3c9d3ffd.svg"
                />
              </i>
            </li>
          </div>
          <span
            class="tooltip-text"
          >
            Add to your watchlist
          </span>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div>
            <li
              class="option-single"
            >
              <i>
                <img
                  alt="Your rating"
                  src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-49-star-white-5c85220678b312aea9599d5f12ad858a9e7df226de51ef8b6b699023ffeda5fa.svg"
                />
              </i>
            </li>
          </div>
          <span
            class="tooltip-text"
          >
            Rate It!
          </span>
        </div>
      </ul>
      <div
        class="overview"
      >
        <h3>
          Overview
        </h3>
        <p
          class="desc"
          data-testid="desc"
        >
          Movie over view
        </p>
      </div>
    </div>
  </section>
</div>
`;
